xbmc/guilib/GUITextLayout.cpp

```diff
@@ -579,63 +579,117 @@ void CGUITextLayout::WrapText(const vecText &text, float maxWidth)
       continue;
     }
 
-    vecText::const_iterator pos = line.m_text.begin();
-    vecText::const_iterator lastBeginPos = line.m_text.begin();
-    vecText::const_iterator lastSpacePos = line.m_text.end();
-    vecText curLine;
+    std::vector<character_t> tmp;
+    tmp.reserve(line.m_text.size());
+    auto widthOf = [&](const std::vector<character_t>::const_iterator start,
+                       const std::vector<character_t>::const_iterator end) -> float
+    {
+      if (start == end)
+        return 0.0f;
+      tmp.assign(start, end);
+      return m_font->GetTextWidth(tmp);
+    };
 
-    while (pos < line.m_text.end())
+    auto skipLeadingSpaces = [&](vecText::const_iterator& it)
     {
-      // Get the current letter in the string
-      const character_t& letter = *pos;
+      while (it != line.m_text.end() && CanWrapAtLetter(*it))
+        ++it;
```

Comment:
This loop could be replaced with an algorithm:
```suggestion
      it = std::find_if_not(it, line.m_text.end(),
                            std::bind_front(&CGUITextLayout::CanWrapAtLetter, this));
```